evalml/pipelines/components/transformers/column_selectors.py

```diff
@@ -122,3 +125,41 @@ def transform(self, X, y=None):
             pd.DataFrame: Transformed X.
         """
         return super().transform(X, y)
+
+
+class SelectByType(ColumnSelector):
+    """
+    Selects columns by specified Woodwork logical type or semantic tag in input data.
+
+    Arguments:
+        column_types (string, ww.LogicalType, list(string), list(ww.LogicalType)): List of Woodwork types or tags, used to determine which columns to select.
+        random_seed (int): Seed for the random number generator. Defaults to 0.
+    """
+
+    name = "Select Columns By Type Transformer"
+    hyperparameter_ranges = {}
+    """{}"""
+    needs_fitting = False
+    _VALIDATION_ERROR_FORMAT_STRING = (
+        "Columns of type {column_types} not found in input data."
+    )
+
+    def __init__(self, column_types=None, random_seed=0, **kwargs):
+        parameters = {"column_types": column_types}
+        parameters.update(kwargs)
+        Transformer.__init__(
+            Transformer,
+            parameters=parameters,
+            component_obj=None,
+            random_seed=random_seed,
+        )
+
+    def _modify_columns(self, cols, X, y=None):
+        return X.ww.select(cols)
+
+    def transform(self, X, y=None):
+        X = infer_feature_types(X)
+        self._check_input_for_columns(X)
```

Comment:
Interesting... does `_check_input_for_columns` do anything here? Since we don't define it, I'm assuming it uses `ColumnSelector`'s, which has the line:
```
        cols = self.parameters.get("columns") or []
```
That returns an empty list for this component and therefore doesn't do any checks, right?

Reply:
It's true that that line will return an empty list and therefore the missing column name check will do nothing - that's by design here. This new bit of code should check if there are column types and if they match though:
```
col_types = self.parameters.get("column_types")
if col_types:
       missing_cols = X.ww.select(col_types).empty
```
I agree, it does feel a little weird to do this here, I'm thinking it may make more sense to move this to `check_input_for_columns` within `SelectByType` itself.